src/appwindow.cpp

```diff
@@ -954,6 +964,11 @@ void AppWindow::onEditorTextChanged(MainWindow *window)
             lspTimerJava->start();
         else
             lspTimerPython->start();
+
+        if (SettingsHelper::isWakaTimeEnable())
+        {
+            wakaTime->sendHeartBeat(window->getFilePath(), true);
+        }
```

Comment:
Could become very expensive since this callback is called for every character pressed inside editor.

Reply:
But only a few operations are done. API requests are not always sent.

Reply:
Yes, I see it, the name should be changed to `wakatime->maybeSendHeartbeat(..)` because it doesn't always sends heartbeat.